src/main/java/seedu/address/logic/parser/CheckInCommandParser.java

```diff
@@ -0,0 +1,55 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_END_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PERSONAL_ID;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ROOM_ID;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_START_DATE;
+
+import java.time.LocalDate;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.CheckInCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new {@code CheckInCommand} object
+ */
+public class CheckInCommandParser implements Parser<CheckInCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the {@code CheckInCommand}
+     * and returns a {@code CheckInCommand} object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public CheckInCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_PERSONAL_ID, PREFIX_ROOM_ID,
+                PREFIX_START_DATE, PREFIX_END_DATE);
+
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_PERSONAL_ID, PREFIX_ROOM_ID, PREFIX_START_DATE, PREFIX_END_DATE)
+            || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, CheckInCommand.MESSAGE_USAGE));
+        }
+
+        int personalId = ParserUtil.parsePersonalId(argMultimap.getValue(PREFIX_PERSONAL_ID).get());
+        int roomId = ParserUtil.parseRoomId(argMultimap.getValue(PREFIX_ROOM_ID).get());
```

Comment:
Just want to double check, the checking of the length of the roomId is done in the execute which is not yet implemented right?

Reply:
as what @shermz-lim mentioned earlier during our meeting, he suggested that there shouldn't be a checking of length for the room id. if the user keys in roomId of length of 5, for example, the user will just be greeted with a 'room does not exist'